### PR TITLE
Canonicalizes Spaces in Semicolon Stringlists, fixes 3086

### PR DESCRIPTION
**Pull request by daniel-j-h**
# Issue

For #3086.

```
Spaces;











  ;  Spaces Everywhere



























Remove; All; Of; Them
```

## Tasklist
 - [x] review
 - [x] adjust for for comments

### CODE REVIEW

File: profiles/lib/stringlist.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
+local Stringlist = {}
+
+-- Takes a string representing a list separated by delim and canonicalizes containing spaces.
+--
+-- Example:
+-- In: aaa;bbb; ccc;  d;dd"
+-- Out: aaa; bbb; ccc; d; dd"
+function Stringlist.canonicalize_spaces(str, delim)
```
Comment (daniel-j-h) [inline profiles/lib/stringlist.lua]: The following is the least worst Lua I could come up with to solve this...

File: profiles/car.lua [hunk 2]
```diff
@@ -3,6 +3,7 @@ local find_access_tag = require("lib/access").find_access_tag
 local get_destination = require("lib/destination").get_destination
 local set_classification = require("lib/guidance").set_classification
 local get_turn_lanes = require("lib/guidance").get_turn_lanes
+local canonicalize_spaces = require("lib/stringlist").canonicalize_spaces
```
Comment (TheMarex) [inline profiles/car.lua]: This is a new dependency for `car.lua` and would impact downstream consumers. It needs a changelog entry.

File: profiles/lib/stringlist.lua (hunk 1 above)
Comment (TheMarex) [inline profiles/lib/stringlist.lua]: Devils advocate here: Why not do this in C++-land and expose a helper function to lua?

Comment (daniel-j-h) [inline profiles/lib/stringlist.lua]: Done - can you review again, and merge if it's okay with you.

**Comment by TheMarex** [comment]
Looks good. :+1: Needs a rebase.

**Pull request merged**